infra/modules/aws-ssm-secrets/main.tf

```diff
@@ -11,13 +11,15 @@ locals {
   path_prefix              = local.non_empty_path && local.non_fully_qualified_path ? "/${var.path}" : var.path
 }
 
+# see: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ssm_parameter
 resource "aws_ssm_parameter" "secret" {
   for_each = var.secrets
 
   name        = "${local.path_prefix}${each.key}"
   type        = "SecureString"
   description = each.value.description
   value       = sensitive(each.value.value)
+  key_id      = coalesce(var.kms_key_id, "alias/aws/ssm")
```

Comment:
Where does "alias/aws/ssm" comes from??? it is optional, why not leaving it empty?

Reply:
weirdly, `coalesce` throws error if all `null`/blank - https://developer.hashicorp.com/terraform/language/functions/coalesce